truffle/src/main/java/org/jruby/truffle/nodes/ext/BigDecimalNodes.java

```diff
@@ -122,6 +133,28 @@ public static RubyBasicObject createRubyBigDecimal(RubyClass rubyClass, BigDecim
         return new RubyBasicObject(rubyClass, BIG_DECIMAL_FACTORY.newInstance(Type.NORMAL, value));
     }
 
+    public static RoundingMode toRoundingMode(int constValue) {
+        switch (constValue) {
+            case 1:
+                return RoundingMode.UP;
+            case 2:
+                return RoundingMode.DOWN;
+            case 3:
+                return RoundingMode.HALF_UP;
+            case 4:
+                return RoundingMode.HALF_DOWN;
+            case 5:
+                return RoundingMode.CEILING;
+            case 6:
+                return RoundingMode.FLOOR;
+            case 7:
+                return RoundingMode.HALF_EVEN;
+        }
+
+        CompilerAsserts.neverPartOfCompilation();
+        throw new UnsupportedOperationException();
```

Comment:
If you want this to never be part of compilation, you might just want to add a `@TruffleBoundary` on the method.  But I suspect you only want the failure case to not be part of compilation. In which case I think you might want to just do a transfer.